gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -449,6 +455,10 @@ def self.install_hooks(mod)
       return
     end
 
+    if !mod.is_a?(Module)
+      return
+    end
```

Comment:
We should not land this! This was a hack I did to make the logs less chunderous.

Reply:
If this did not cause a test to fail, please add a test so that this for sure fails.

Reply:
Making this `mod = mod.singleton_class if !mod.is_a?(Module)` leads to equivalent results (and should perhaps be moved up a bit if we decided to go that route).  Removing this bit leads to problems with custom decorators and props.

Reply:
Yes I'm saying that those problems are real final violations, and when I was looking into those I knew that we'd have to fix all of those, and wanted to see the rest of the errors, so I skipped them with this line, knowing that we'd have to go back and fix those violations too.

Reply:
So, `mod = mod.singleton_class if !mod.is_a?(Module)` is probably wrong, but the early return here is not.  Actually, the early return should probably be quite a bit earlier so that we don't shove non-`Module` things into `@installed_hooks`.  The sequence of events is:

1. A class including `T::Props` (the particular instance has a custom decorator), `Model`, includes a module that defines `ClassMethods`.  Call this included module G.
2. We call `T::Props::ClassMethods#included`, passing `Model` as the child.

https://github.com/sorbet/sorbet/blob/88e72f6da215d2952b761b0f4398d4296bfc16e3/gems/sorbet-runtime/lib/types/props/_props.rb#L141-L144

3. We, via `super`, call `T::Props::Plugin::ClassMethods#included`.

https://github.com/sorbet/sorbet/blob/88e72f6da215d2952b761b0f4398d4296bfc16e3/gems/sorbet-runtime/lib/types/props/plugin.rb#L8-L13

4. The call to `plugin` resolves to the mixed-in `T::Props::ClassMethods#plugin`; `mod` is G.

https://github.com/sorbet/sorbet/blob/88e72f6da215d2952b761b0f4398d4296bfc16e3/gems/sorbet-runtime/lib/types/props/_props.rb#L122-L125

5. The call to `plugin` resolves to `T::Props::Decorator::Plugin#plugin`.

https://github.com/sorbet/sorbet/blob/88e72f6da215d2952b761b0f4398d4296bfc16e3/gems/sorbet-runtime/lib/types/props/decorator.rb#L670-L675

6.  We call `T::Props::Plugin::Private.apply_decorator_methods(mod, self)`.  Note that `self` here is an *instance* of the decorator.

https://github.com/sorbet/sorbet/blob/88e72f6da215d2952b761b0f4398d4296bfc16e3/gems/sorbet-runtime/lib/types/props/plugin.rb#L29-L35

7. `target` is an instance of our decorator and `plugin` is G, above, so we call `decorator.extend(G::DecoratorMethods)`
8. We wind up in our replaced method hook for `extended`:

https://github.com/sorbet/sorbet/blob/88e72f6da215d2952b761b0f4398d4296bfc16e3/gems/sorbet-runtime/lib/types/private/methods/_methods.rb#L408-L411

9. `arg` is an instance of the decorator and `self` is `G::DecoratorMethods`.  We think `arg` is a module, but that is not necessarily the case.
10. We call `_hook_impl(arg, arg.singleton_class.ancestors, self)`.

https://github.com/sorbet/sorbet/blob/88e72f6da215d2952b761b0f4398d4296bfc16e3/gems/sorbet-runtime/lib/types/private/methods/_methods.rb#L384-L393

11. We call `install_hooks(target)`, and `install_hooks` is receiving an instance, not a `Module`.

This is all a bit deep in the metaprogramming weeds for me, but I am loath to tug on any piece of this.  If I had to pull on something, I'd guess that the `apply_decorator_methods` call/code needs to be modified, but I am completely unsure of how.